adr/ADR-XX.md

```diff
@@ -0,0 +1,76 @@
+# Subject Mapping Transforms in Streams
+
+|Metadata| Value                     |
+|--------|---------------------------|
+|Date    | 2023-02-10                |
+|Author  | @jnmoyne                  |
+|Status  | `Partially Implemented`   |
+|Tags    | jetstream, client, server |
+
+## Context and Problem Statement
+
+Subject mapping and transformation is only available at the Core NATS level, meaning that in order to define or modify mappings one has to either have access to the server config file, or have access to the account's key in operator security mode. While Core NATS subject mapping has its place and use (e.g. scaling a single stream for writes using partitioning), many (most) use cases for subject mapping happen in the context of streams, and having to go to the Core NATS server/account level to define subject mappings is quite limiting as it's not easy for an application programmer to be able to define the mappings he/she needs (even if they have access to the account's key).
```

Comment:
Have we officially defined **Core NATS** and institutionalized?

Reply:
Yes as far as I know, there are certainly a number of references to "Core NATS" in the docs (to distinguish with JS)